torch/distributed/tensor/_ops/_einsum_strategy.py

```diff
@@ -140,6 +148,8 @@ def gen_einsum_strategies(
         # split rhs free dim
         for rhs_dim in edims.rhs_out_only_dims:
             rhs_free_dim = output_dim.index(rhs_dim)
+            if rhs_free_dim >= edims.input_ndim[1]:
```

Comment:
IMO we can do this simpler without recording the `input_ndim` list. We should distinguish `rhs_free_dim_output` and `rhs_free_dim_input`, and `rhs_free_dim_input = input_dims[1].index(rhs_dim)`

Then the rhs_placement_list should be adjusted correspondingly.